### PR TITLE
Avoid installing ChainerX when building docs of other projects on ReadTheDocs

### PR DESCRIPTION
**Pull request by knorth55**
install `chainerx` when `READTHEDOCS_PROJECT` name is `chainer`.
this PR stop installing `chainerx` in other projects depending chainer.

### CODE REVIEW

**Comment by knorth55** [comment]
https://readthedocs.org/projects/chainercv/builds/9156363/
chainercv doc build failing because of this problem.

File: setup.py [hunk 1]
```diff
@@ -186,7 +186,8 @@ def find_any_distribution(pkgs):
 
 
 build_chainerx = 0 != int(os.getenv('CHAINER_BUILD_CHAINERX', '0'))
-if os.getenv('READTHEDOCS', None) == 'True':
+if os.getenv('READTHEDOCS', None) == 'True' \
```
Comment (kmaehashi) [inline setup.py]: Could you surround conditions `()` and avoid using `\`?

Comment (knorth55) [reply setup.py]: I updated, thank you.

**Comment by knorth55** [comment]
https://knorth55-chainer.readthedocs.io/en/fix-readthedocs/
doc build succeeded.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by knorth55** [comment]
jenkins test failed, but it seems it caught unrelated test failure.
https://jenkins.preferred.jp/job/chainer/job/chainer_pr/1151/TEST=chainer-py2,label=mn1-p100/testReport/tests.chainer_tests.functions_tests.connection_tests.test_convolution_nd/TestConvolutionND_use_chainerx_false__chainerx_device_None__use_cuda_true__cuda_device_0__use_cudnn_always__cudnn_deterministic_false__autotune_false__cudnn_fast_batch_normalization_false__use_ideep_never_param_81_%7Bb_dtype=float32,%20dims=(3,%204,%203),%20autotune=False,%20cover_all=False,%20groups=1,%20dilate=1,%20nobias=False,%20contiguous='C',%20W_dtype=float32,%20x_dtype=float32%7D/test_backward/

**Pull request merged**